### PR TITLE
Simplify the bitfield signing job

### PR DESCRIPTION
**Pull request by bkchr**
Besides that the pr also adds a simple test.

### CODE REVIEW

File: node/core/bitfield-signing/src/lib.rs [hunk 1]
```diff
@@ -181,85 +169,55 @@ async fn get_core_availability(
...
-				Err(err) => {
-					errs_ref.lock().await.push(err);
-					return;
-				}
-			};
-			out_ref.lock().await.set(idx, availability);
-		})
-		.await;
+	// Wrap the sender in a Mutex to share it between the futures.
+	let sender = Mutex::new(sender);
```
Comment (montekki) [inline node/core/bitfield-signing/src/lib.rs:212]: Why not `clone()`?

**Review by montekki — APPROVED** [review]

**Review by coriolinus — APPROVED** [review]

File: node/core/bitfield-signing/src/lib.rs (hunk 1 above)
Comment (drahnr) [reply node/core/bitfield-signing/src/lib.rs:212]: That would increase the capacity by 1 for each clone.

Comment (bkchr) [reply node/core/bitfield-signing/src/lib.rs:212]: Exactly, this would mean that you can always send at least one item.

**Comment by montekki** [comment]
bot merge

File: node/core/bitfield-signing/src/lib.rs (hunk 1 above)
Comment (pepyakin) [reply node/core/bitfield-signing/src/lib.rs:212]: Could one of you guys please send a follow-up to leave a comment for this? I think it is important.

**Pull request merged**